pkgs/development/tools/jf/default.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, stdenv, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "jf";
+  version = "0.2.2";
+
+  src = fetchFromGitHub {
+    owner = "sayanarijit";
+    repo = pname;
```

Comment:
```suggestion
    repo = "jf";
```

Reply:
There are many packages with `repo = pname`... Any particular reason not to use `pname`?

Reply:
> There are many packages with `repo = pname`...

It implies they should be fixed too.
(Indeed this is on my backlog; I'll do this after or in parallel to removing `with lib;` occurrences...)

> Any particular reason not to use `pname`?

Ideally `pname` never changes, therefore there is no reason to parameterize it.
This would be equivalent to write `1+1` instead of `2`.

Further, it is not uncommon that `pname` can't be used directly as is.
E.G. when `src.repo` uses a mix of lowercase and uppercase letters (like openMSX).

So, what we should do? Write a complex expression that converts `pname` to `src.repo`?
Obviously no, because `pname` and `src.repo` will never change.
Just treat them as the constants they always were.

Further, it reduces interdependencies in the code.